base/array.jl

```diff
@@ -1441,13 +1436,16 @@ function resize!(a::Vector, nl::Integer)
 end
 
 """
-    sizehint!(s, n) -> s
+    sizehint!(s, n, [first=false]; shink=false) -> s
 
 Suggest that collection `s` reserve capacity for at least `n` elements. That is, if
 you expect that you're going to have to push a lot of values onto `s`, you can avoid
 the cost of incremental reallocation by doing it once up front; this can improve
 performance.
 
+If `first` is true, then the reserved space is from the start of the
+collection, if `pushfirst!` is supported on the collection.
```

Comment:
It could be interesting to widen the scope of the `first` argument, e.g. `sizehint!(::BitSet)` could also benefit from this argument (and just forward it to the underlying vector in the current implementation), but does not support `pushfirst!`.

Reply:
Good point. I changed the documentation to mention ordered collections, so that `pushfirst` is now just an example instead of the spec